src/services/entry-service.ts

```diff
@@ -1,40 +1,79 @@
-import {EntryListResponse} from 'types/entry-list-response';
+import {EntryData, EntryListResponse} from 'types/entry-list-response';
+import {Sources} from 'types/sources';
 import {RelatedLoader} from './related-loader';
 
+import * as humanizeDuration from 'humanize-duration';
+
 class EntryService {
   private player: KalturaPlayerTypes.Player;
   private logger: KalturaPlayerTypes.Logger;
+  private getDurationText: (duration?: number) => string = () => '';
 
   constructor(player: KalturaPlayerTypes.Player, logger: KalturaPlayerTypes.Logger) {
     this.player = player;
     this.logger = logger;
+
+    try {
+      const durationHumanizer = getDurationHumanizer();
+      if (durationHumanizer) {
+        this.getDurationText = (duration?: number) => {
+          try {
+            return duration ? durationHumanizer(duration * 1000) : '';
+          } catch (e: any) {
+            return '';
+          }
+        };
+      }
+    } catch (e: any) {
+      console.log(e);
+    }
   }
 
-  async getByPlaylist(playlistInfo: {playlistId: string; ks?: string}): Promise<KalturaPlayerTypes.Sources[]> {
+  async getByPlaylist(playlistInfo: {playlistId: string; ks?: string}): Promise<Sources[]> {
     try {
       const response: EntryListResponse = await this.player.provider.getPlaylistConfig(playlistInfo);
-      return processResponse(response);
+      return response.items.map(({sources}: EntryData, index) => {
+        return {
+          ...sources,
+          internalIndex: index,
+          durationText: this.getDurationText(sources.duration)
+        };
+      });
     } catch (e) {
       this.logger.warn(`failed to get related entries by playlist id ${playlistInfo.playlistId}`);
       return [];
     }
   }
 
-  async getByEntryList(entryList: {entries: KalturaPlayerTypes.MediaInfo[]; ks?: string}): Promise<KalturaPlayerTypes.Sources[]> {
+  async getByEntryList(entryList: {entries: KalturaPlayerTypes.MediaInfo[]; ks?: string}): Promise<Sources[]> {
     try {
       const response: EntryListResponse = await this.player.provider.getEntryListConfig(entryList);
-      return processResponse(response);
+      return response.items.map(({sources}: EntryData, index) => {
+        return {
+          ...sources,
+          internalIndex: index,
+          durationText: this.getDurationText(sources.duration)
+        };
+      });
     } catch (e) {
       this.logger.warn(`failed to get related entries by entry list`);
       return [];
     }
   }
 
-  getBySourcesList(sourcesList: KalturaPlayerTypes.Sources[]): KalturaPlayerTypes.Sources[] {
-    return sourcesList.filter(sources => this.isPlayable(sources));
+  getBySourcesList(sourcesList: KalturaPlayerTypes.Sources[]): Sources[] {
```

Comment:
```suggestion
  getBySourcesList(sourcesList: Sources[]): Sources[] {
```

Reply:
KalturaPlayerTypes.Sources doesn't have the 2 additional fields